engine/utils/list_tools.py

```diff
@@ -1,12 +1,16 @@
+"""Module containing helper functions for list manipulation."""
+
+
 def flatten(list_of_lists):
     """
-    Flattens a list of list into a single flat list.
+    Flatten a list of list into a single flat list.
 
     Arguments:
         list_of_lists {list[list[T]]} -- List of lists to flatten.
 
     Returns:
         list[T] -- Flat list generated by flattening the source list of lists.
+
     """
     flat = []
 
```

Comment:
chain.from_iterable(listOfLists)

Reply:
I saw `itertools.chain.from_iterable()` on StackOverflow, but I didn't want to add an import just for this, because the performance difference is negligible. Somehow I didn't realize `itertools` are in the standard library. I will change it within this PR. #95